src/PubSub/Drivers/RedisClient.php

```diff
@@ -258,6 +297,19 @@ public function channelMemberCounts($appId, array $channelNames): PromiseInterfa
             });
     }
 
+    /**
+     * Get the amount of unique connections.
+     *
+     * @param  mixed  $appId
+     * @return null|int|\React\Promise\PromiseInterface
+     */
+    public function appConnectionsCount($appId)
+    {
+        // Use the in-built Redis manager to avoid async run.
+
+        return $this->redis->hget($this->getTopicName($appId), 'connections') ?: 0;
```

Comment:
@francislavoie Here I have to use the in-built Redis connection because I need the promise to be resolved. I tried using https://github.com/clue/reactphp-block but i cannot access the loop and it will stop it at some point.

I'll update the docblock accordingly tho.

Reply:
That's still not okay though. Any invocation of the non-reactphp redis client will block the _entire_ websocket server until it's done. Performance and scaling will suffer significantly. You need to return the promise and have whatever calls it resolve the promise as part of the loop. It's an annoying refactor I'm sure, but it's necessary.

Reply:
So one place it's used is here: https://github.com/beyondcode/laravel-websockets/blob/2.x/src/WebSockets/WebSocketHandler.php#L155

I think you should actually have two separate methods - local connection count, and global connection count. For the websocket handler, it should check the local connection count to know whether you hit the limit. The limit shouldn't be global, it should be per server.

For the UI dashboard, you can use the global connection count and resolve the promise before rendering the response.

Reply:
The idea behind it was that if you have multi-node WebSocket servers, you want to enforce the global connection count to be `X`, rather than a per-server connection limit. 🤔 You might want to have a self-hosted, multi-node, WebSockets server that will handle multiple apps and you want to limit them globally.

Reply:
A global limit opens the door for some servers being more overloaded than others, which I don't think makes sense. Instead, you'd want to spread out the load, so per-server limits is more useful.

For example if you have a limit of 100 and 2 servers, one server could be handling 80 connections and the other 20 and it would stop handling new connections while the first is overloaded but the second still has plenty of room. Doesn't make much sense to me.

Also if your limit is global, you prevent scaling unless you also change the limit when scaling up. That's not ideal. If your limit is per server, then you can just spin up a new one to add more room if you get overloaded and no config has to change.

Reply:
This problem can be easily solved by the infrastructure itself by "least connections" or "round-robin". That's why I am all-in into this kind of limitation app-by-app, just like Pusher does. 🤔

Reply:
I agree, which is why I wouldn't actually use the connection limit myself.

I guess I don't understand when a global limit is ever useful compared to a per-server limit. A per-server limit is actually scalable (has a multiplier), that seems much more useful.

Reply:
Then the local limit should be useful for the scaling on each server, while the global limit should be used to enforce a max amount of connections on an app-by-app basis. Think of it like hosting your own server that will hold multiple apps and enforcing a limit for each one.